### PR TITLE
Events API: Transaction dropped, sync status, and renames

### PR DESCRIPTION
**Pull request by RatanRSur**
## PR description
Adds the ability to listen to SyncStatus changes and transaction dropped events to the `PantheonEvents` API

### CODE REVIEW

File: ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/state/SyncState.java [hunk 1]
```diff
@@ -49,17 +51,22 @@ public SyncState(final Blockchain blockchain, final EthPeers ethPeers) {
         });
   }
 
-  private void publishSyncStatus() {
+  @VisibleForTesting
```
Comment (abdelhamidbakhta) [inline ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/state/SyncState.java]: (discussion) : why `@VisibleForTesting` if public ?

File: pantheon/src/test/java/tech/pegasys/pantheon/services/PantheonEventsImplTest.java [hunk 2]
```diff
@@ -166,25 +190,55 @@ public void newTransactionEventFiresAfterSubscribe() {
...
+
+    assertThat(result.get()).isNull();
+    // The max pool size is configured to 1 so adding two transactions should trigger a drop
+    transactionPool.addLocalTransaction(TX1);
+    transactionPool.addLocalTransaction(TX2);
+
+    assertThat(result.get()).isNotNull();
+  }
+
+  @Test
```
Comment (abdelhamidbakhta) [inline pantheon/src/test/java/tech/pegasys/pantheon/services/PantheonEventsImplTest.java]: nice test 🎉

File: ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/state/SyncState.java (hunk 1 above)
Comment (RatanRSur) [reply ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/state/SyncState.java]: I didn't know this before but `@VisibleForTesting` is for documentation purposes only! I used to think it actually changed the visibility for the tests in some special way. It's to communicate that if not for the tests, this method would be less visible.

File: ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/health/ReadinessCheckTest.java [hunk 3]
```diff
@@ -141,6 +141,7 @@ public void shouldNotBeReadyWhenCustomMaxBlocksBehindIsInvalid() {
   }
 
   private Optional<SyncStatus> createSyncStatus(final int currentBlock, final int highestBlock) {
-    return Optional.of(new SyncStatus(0, currentBlock, highestBlock));
+    return Optional.of(
+        new tech.pegasys.pantheon.ethereum.core.SyncStatus(0, currentBlock, highestBlock));
```
Comment (shemnon) [inline ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/health/ReadinessCheckTest.java]: If only java had type aliases on import...

**Pull request merged**